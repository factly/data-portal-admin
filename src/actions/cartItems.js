import axios from '../utils/axios';
import {
  ADD_CARTITEMS,
  SET_CARTITEM_LOADING,
  SET_CARTITEM_REQUEST,
  SET_CARTITEM_IDS,
  CARTITEM_API,
} from '../constants/cartItems';
import { addProducts } from './products';
import { getIds, getValues, deleteKeys, buildObjectOfItems } from '../utils/objects';

export const loadCartItems = (id, page = 1, limit = 5) => {
  return async (dispatch, getState) => {
    const {
      cartItems: { req },
    } = getState();

    let ids;
    for (let item of req) {
      if (item.page === page && item.limit === limit) {
        ids = [...item.ids];
      }
    }

    if (ids) {
      dispatch(setCartItemIds(ids));
      return;
    }

    dispatch(setLoading(true));

    const response = await axios({
      url: `${CARTITEM_API(id)}?page=${page}&limit=${limit}`,
      method: 'get',
    });

    const { nodes, total } = response.data;
    const currentPageIds = getIds(nodes);
    const currentReq = { id, page: page, limit: limit, ids: currentPageIds };
    dispatch(setCartItemRequest(currentReq, total));
    dispatch(addCartItems(nodes));
    dispatch(setCartItemIds(currentPageIds));

    dispatch(setLoading(false));
  };
};

const setLoading = (loading) => {
  return {
    type: SET_CARTITEM_LOADING,
    payload: { loading },
  };
};

export const addCartItems = (cartItems) => (dispatch) => {
  const products = getValues(cartItems, 'product');
  dispatch(addProducts(products));

  dispatch({
    type: ADD_CARTITEMS,
    payload: {
      cartItems: buildObjectOfItems(deleteKeys(cartItems, ['product'])),
    },
  });
};

const setCartItemRequest = (req, total) => {
  return {
    type: SET_CARTITEM_REQUEST,
    payload: { req, total },
  };
};

const setCartItemIds = (ids) => {
  return {
    type: SET_CARTITEM_IDS,
    payload: { ids },
  };
};
