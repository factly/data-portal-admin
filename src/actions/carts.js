import axios from '../utils/axios';
import {
  baseUrl,
  LOADING_CARTS,
  LOADING_CART_DETAILS,
  LOAD_CARTS_SUCCESS,
  LOAD_CARTS_FAILURE,
  GET_CART_ITEMS_SUCCESS,
  GET_CART_ITEMS_FAILURE,
} from '../constants/carts';

export const loadCarts = (page, limit) => {
  return async (dispatch, getState) => {
    let url = baseUrl;
    if (page && limit) {
      url = `${url}?page=${page}&limit=${limit}`;
    }

    dispatch(loadingCarts());

    const response = await axios({
      url: url,
      method: 'get',
    }).catch((error) => {
      dispatch(loadCartsFailure(error.message));
    });

    if (response) {
      dispatch(loadCartsSuccess(response.data));
    }
  };
};

export const getCartItems = (id, page, limit) => {
  return async (dispatch, getState) => {
    let url = `${baseUrl}/${id}/items`;
    if (page && limit) {
      url = `${url}?page=${page}&limit=${limit}`;
    }

    dispatch(loadingCartDetails());

    const response = await axios({
      url: url,
      method: 'get',
    }).catch((error) => {
      dispatch(getCartItemsFailure(error.message));
    });

    if (response) {
      dispatch(getCartItemsSuccess(response.data));
    }
  };
};

const loadingCarts = () => {
  return {
    type: LOADING_CARTS,
  };
};

const loadingCartDetails = () => {
  return {
    type: LOADING_CART_DETAILS,
  };
};

const loadCartsSuccess = (data) => {
  return {
    type: LOAD_CARTS_SUCCESS,
    payload: {
      items: data.nodes,
      total: data.total,
    },
  };
};

const loadCartsFailure = (message) => {
  return {
    type: LOAD_CARTS_FAILURE,
    payload: message,
  };
};

const getCartItemsSuccess = (data) => {
  return {
    type: GET_CART_ITEMS_SUCCESS,
    payload: {
      items: data.nodes,
      total: data.total,
    },
  };
};

const getCartItemsFailure = (message) => {
  return {
    type: GET_CART_ITEMS_FAILURE,
    payload: message,
  };
};
