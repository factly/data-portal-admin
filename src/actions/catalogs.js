import axios from '../utils/axios';
import {
  baseUrl,
  ADD_CATALOGS_LIST_REQUEST,
  SET_CATALOGS_LIST_CURRENT_PAGE,
  GET_CATALOG_SUCCESS,
  GET_CATALOG_FAILURE,
  LOADING_CATALOGS,
  LOAD_CATALOGS_SUCCESS,
  SET_CATALOGS_LIST_TOTAL,
  LOAD_CATALOGS_FAILURE,
  CREATING_CATALOG,
  CREATE_CATALOG_SUCCESS,
  CREATE_CATALOG_FAILURE,
  UPDATING_CATALOG,
  UPDATE_CATALOG_SUCCESS,
  UPDATE_CATALOG_FAILURE,
  DELETING_CATALOG,
  DELETE_CATALOG_SUCCESS,
  DELETE_CATALOG_FAILURE,
} from '../constants/catalogs';
import { loadProductsSuccess } from './products';
import { getIds, buildObjectOfItems, getValues } from '../utils/objects';

export const loadCatalogs = (page, limit) => {
  return async (dispatch, getState) => {
    let url = baseUrl;
    if (page && limit) {
      url = `${url}?page=${page}&limit=${limit}`;
    }

    const {
      catalogs: { req },
    } = getState();

    let found = false;
    let ids;
    for (let item of req) {
      if (item.page === page && item.limit === limit) {
        ids = [...item.ids];
        found = true;
      }
    }

    if (found) {
      dispatch(setListCurrentPage(ids));
      return;
    }

    dispatch(loadingCatalogs());

    const response = await axios({
      url: url,
      method: 'get',
    }).catch((error) => {
      dispatch(loadCatalogsFailure(error.message));
    });

    if (response) {
      const { nodes, total } = response.data;
      const currentPageIds = getIds(nodes);
      const req = { page: page, limit: limit, ids: currentPageIds };

      const products = getValues(nodes, 'products').filter((product) => product);
      products && dispatch(loadProductsSuccess(products));

      nodes.forEach((catalog) => {
        catalog.products = getIds(catalog.products || []);
      });

      dispatch(addListRequest(req));
      dispatch(loadCatalogsSuccess(nodes));
      dispatch(setListCurrentPage(currentPageIds));
      dispatch(setCatalogsListTotal(total));
    }
  };
};

export const createCatalog = (data) => {
  return async (dispatch, getState) => {
    dispatch(creatingCatalog());

    const response = await axios({
      url: baseUrl,
      method: 'post',
      data: data,
    }).catch((error) => {
      dispatch(createCatalogFailure(error.message));
    });

    if (response) {
      dispatch(createCatalogSuccess(response.data));
    }
  };
};

export const updateCatalog = (id, data) => {
  return async (dispatch, getState) => {
    let url = `${baseUrl}/${id}`;

    dispatch(updatingCatalog());

    const response = await axios({
      url: url,
      method: 'put',
      data: data,
    }).catch((error) => {
      dispatch(updateCatalogFailure(error.message));
    });

    if (response) {
      dispatch(updateCatalogSuccess(response.data));
    }
  };
};

export const deleteCatalog = (id) => {
  return async (dispatch, getState) => {
    let url = `${baseUrl}/${id}`;

    dispatch(deletingCatalog());

    const response = await axios({
      url: url,
      method: 'delete',
    }).catch((error) => {
      dispatch(deleteCatalogFailure(error.message));
    });

    if (response) {
      dispatch(deleteCatalogSuccess(id));
    }
  };
};

export const getCatalog = (id) => {
  return async (dispatch, getState) => {
    const {
      catalogs: { items },
    } = getState();

    if (items[id]) {
      const catalog = { ...items[id] };
      dispatch(getCatalogSuccess(catalog));
      return;
    }

    dispatch(loadingCatalogs());

    const response = await axios({
      url: `${baseUrl}/${id}`,
      method: 'get',
    }).catch((error) => {
      dispatch(getCatalogFailure(error.message));
    });

    if (response) {
      const catalog = response.data;
      catalog.products = getIds(catalog.products || []);
      dispatch(getCatalogSuccess(catalog));
    }
  };
};

const loadingCatalogs = () => {
  return {
    type: LOADING_CATALOGS,
  };
};

const setCatalogsListTotal = (total) => {
  return {
    type: SET_CATALOGS_LIST_TOTAL,
    payload: total,
  };
};

const addListRequest = (req) => {
  return {
    type: ADD_CATALOGS_LIST_REQUEST,
    payload: req,
  };
};

const setListCurrentPage = (ids) => {
  return {
    type: SET_CATALOGS_LIST_CURRENT_PAGE,
    payload: ids,
  };
};

const getCatalogSuccess = (catalog) => {
  return {
    type: GET_CATALOG_SUCCESS,
    payload: catalog,
  };
};

const getCatalogFailure = (message) => {
  return {
    type: GET_CATALOG_FAILURE,
    payload: message,
  };
};

export const loadCatalogsSuccess = (catalogs) => {
  return {
    type: LOAD_CATALOGS_SUCCESS,
    payload: {
      items: buildObjectOfItems(catalogs),
    },
  };
};

const loadCatalogsFailure = (message) => {
  return {
    type: LOAD_CATALOGS_FAILURE,
    payload: message,
  };
};

const creatingCatalog = () => {
  return {
    type: CREATING_CATALOG,
  };
};

const createCatalogSuccess = (catalog) => {
  return {
    type: CREATE_CATALOG_SUCCESS,
    payload: catalog,
  };
};

const createCatalogFailure = (message) => {
  return {
    type: CREATE_CATALOG_FAILURE,
    payload: message,
  };
};

const updatingCatalog = () => {
  return {
    type: UPDATING_CATALOG,
  };
};

const updateCatalogSuccess = (catalog) => {
  return {
    type: UPDATE_CATALOG_SUCCESS,
    payload: catalog,
  };
};

const updateCatalogFailure = (message) => {
  return {
    type: UPDATE_CATALOG_FAILURE,
    payload: message,
  };
};

const deletingCatalog = () => {
  return {
    type: DELETING_CATALOG,
  };
};

const deleteCatalogSuccess = (id) => {
  return {
    type: DELETE_CATALOG_SUCCESS,
    payload: id,
  };
};

const deleteCatalogFailure = (message) => {
  return {
    type: DELETE_CATALOG_FAILURE,
    payload: message,
  };
};
