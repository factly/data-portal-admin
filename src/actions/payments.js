import axios from '../utils/axios';
import {
  ADD_PAYMENTS,
  SET_PAYMENT_LOADING,
  SET_PAYMENT_REQUEST,
  SET_PAYMENT_IDS,
  PAYMENT_API,
} from '../constants/payments';
import { addCurrencies } from './currencies';
import { getIds, getValues, deleteKeys, buildObjectOfItems } from '../utils/objects';

export const loadPayments = (page = 1, limit = 5) => {
  return async (dispatch, getState) => {
    dispatch(setLoading(true));

    const response = await axios({
      url: `${PAYMENT_API}?page=${page}&limit=${limit}`,
      method: 'get',
    });

    const { nodes, total } = response.data;
    const currentPageIds = getIds(nodes);
    const currentReq = { page: page, limit: limit, ids: currentPageIds };
    dispatch(setPaymentRequest(currentReq, total));
    dispatch(addPayments(nodes));
    dispatch(setPaymentIds(currentPageIds));

    dispatch(setLoading(false));
  };
};

export const setLoading = (loading) => {
  return {
    type: SET_PAYMENT_LOADING,
    payload: { loading },
  };
};

export const addPayments = (payments) => (dispatch) => {
  const currencies = getValues(payments, 'currency');
  dispatch(addCurrencies(currencies));

  dispatch({
    type: ADD_PAYMENTS,
    payload: {
      payments: buildObjectOfItems(deleteKeys(payments, ['currency'])),
    },
  });
};

export const setPaymentRequest = (req, total) => {
  return {
    type: SET_PAYMENT_REQUEST,
    payload: { req, total },
  };
};

export const setPaymentIds = (ids) => {
  return {
    type: SET_PAYMENT_IDS,
    payload: { ids },
  };
};
