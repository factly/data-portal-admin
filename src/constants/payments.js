export const LOADING_PAYMENTS = 'LOADING_PAYMENTS';
export const LOAD_PAYMENTS_SUCCESS = 'LOAD_PAYMENTS_SUCCESS';
export const LOAD_PAYMENTS_FAILURE = 'LOAD_PAYMENTS_FAILURE';

export const baseUrl = `${process.env.REACT_APP_API_URL}/payments`;
