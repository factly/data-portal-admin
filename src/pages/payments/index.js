import React from 'react';

import PaymentsList from './components/PaymentsList';

const Payments = () => {
  return <PaymentsList />;
};

export default Payments;
