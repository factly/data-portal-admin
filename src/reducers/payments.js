import {
  LOADING_PAYMENTS,
  LOAD_PAYMENTS_SUCCESS,
  LOAD_PAYMENTS_FAILURE,
  SET_PAYMENTS_LIST_TOTAL,
} from '../constants/payments';
import { unique } from '../utils/objects';

const initialState = {
  list: { loading: false, ids: [], items: {}, total: 0 },
};

export default function paymentsReducer(state = initialState, action = {}) {
  switch (action.type) {
    case LOADING_PAYMENTS:
      return {
        ...state,
        list: {
          ...state.list,
          loading: true,
        },
      };
    case LOAD_PAYMENTS_SUCCESS: {
      const { ids, items } = action.payload;
      const { list } = state;
      return {
        ...state,
        list: {
          ...list,
          loading: false,
          ids: unique([...list.ids, ...ids]),
          items,
        },
      };
    }
    case SET_PAYMENTS_LIST_TOTAL:
      return {
        ...state,
        list: {
          ...state.list,
          total: action.payload,
        },
      };
    case LOAD_PAYMENTS_FAILURE:
      return {
        ...state,
        list: {
          ...state.list,
          loading: false,
        },
      };
    default:
      return state;
  }
}
