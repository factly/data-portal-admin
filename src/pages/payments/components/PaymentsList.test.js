import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer, { act } from 'react-test-renderer';
import { useSelector, useDispatch } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { Table } from 'antd';

import '../../../matchMedia.mock';
import PaymentList from './PaymentsList';
import { loadPayments } from '../../../actions/payments';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
jest.mock('../../../actions/payments', () => ({
  loadPayments: jest.fn(),
  deletePayment: jest.fn(),
}));
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
}));

describe('Payments List component', () => {
  let store;
  let mockedDispatch;
  const payment = {
    id: 1,
    amount: 1,
    currency_id: 1,
    status: 'status',
    gateway: 'Gateway',
    created_at: '2020-12-12',
  };

  describe('snapshot testing', () => {
    beforeEach(() => {
      store = mockStore({});
      store.dispatch = jest.fn();
      mockedDispatch = jest.fn();
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should render the component', () => {
      useSelector.mockImplementation((state) => ({}));
      const tree = renderer.create(<PaymentList />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(useSelector).toHaveBeenCalled();
    });
    it('should match component when loading', () => {
      useSelector.mockImplementation((state) => ({
        data: [],
        total: 0,
        currencies: {},
      }));
      const tree = renderer.create(<PaymentList />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(useSelector).toHaveBeenCalled();
    });
    it('should match component with payments', () => {
      loadPayments.mockReset();
      useSelector.mockImplementation((state) => ({
        data: [payment],
        total: 1,
        currencies: { 1: { id: 1, iso_code: 'INR' } },
      }));

      let component;
      act(() => {
        component = renderer.create(
          <Router>
            <PaymentList />
          </Router>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      expect(useSelector).toHaveBeenCalled();
      expect(mockedDispatch).toHaveBeenCalledTimes(1);
      expect(useSelector).toHaveReturnedWith({
        data: [payment],
        total: 1,
        currencies: { 1: { id: 1, iso_code: 'INR' } },
      });
      expect(loadPayments).toHaveBeenCalledWith(1, 5);
    });
  });
  describe('component testing', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      mockedDispatch = jest.fn(() => new Promise((resolve) => resolve(true)));
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should change the page', (done) => {
      useSelector.mockImplementation((state) => ({}));

      const wrapper = mount(<PaymentList />);
      const table = wrapper.find(Table);
      table.props().pagination.onChange(2, 5);
      wrapper.update();
      const updatedTable = wrapper.find(Table);
      expect(updatedTable.props().pagination.current).toEqual(2);
      setTimeout(() => {
        expect(loadPayments).toHaveBeenLastCalledWith(2, 5);
        done();
      }, 0);
    });
  });
});
