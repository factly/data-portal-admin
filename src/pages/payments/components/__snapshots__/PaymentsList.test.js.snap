// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payments List component snapshot testing should match component when loading 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-bordered"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={Object {}}
          >
            <table
              style={
                Object {
                  "tableLayout": "auto",
                }
              }
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": "25%",
                      "width": "25%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "25%",
                      "width": "25%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "25%",
                      "width": "25%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "25%",
                      "width": "25%",
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Amount
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Gateway
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Status
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Created At
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={1}
                  onClick={[Function]}
                  style={Object {}}
                >
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <span>
                      1 INR
                    </span>
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Gateway
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    status
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <span
                      title="2020-12-12"
                    >
                      4 months ago
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        className="ant-pagination ant-table-pagination ant-table-pagination-right"
        style={Object {}}
        unselectable="unselectable"
      >
        <li
          aria-disabled={true}
          className="ant-pagination-prev ant-pagination-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Previous Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="left"
              className="anticon anticon-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={1}
        >
          <a
            rel="nofollow"
          >
            1
          </a>
        </li>
        <li
          aria-disabled={true}
          className="ant-pagination-next ant-pagination-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Next Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="right"
              className="anticon anticon-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Payments List component snapshot testing should match component with payments 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-bordered"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={Object {}}
          >
            <table
              style={
                Object {
                  "tableLayout": "auto",
                }
              }
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": "25%",
                      "width": "25%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "25%",
                      "width": "25%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "25%",
                      "width": "25%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "25%",
                      "width": "25%",
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Amount
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Gateway
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Status
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Created At
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={1}
                  onClick={[Function]}
                  style={Object {}}
                >
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <span>
                      1 INR
                    </span>
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Gateway
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    status
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <span
                      title="2020-12-12"
                    >
                      4 months ago
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        className="ant-pagination ant-table-pagination ant-table-pagination-right"
        style={Object {}}
        unselectable="unselectable"
      >
        <li
          aria-disabled={true}
          className="ant-pagination-prev ant-pagination-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Previous Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="left"
              className="anticon anticon-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={1}
        >
          <a
            rel="nofollow"
          >
            1
          </a>
        </li>
        <li
          aria-disabled={true}
          className="ant-pagination-next ant-pagination-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Next Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="right"
              className="anticon anticon-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Payments List component snapshot testing should render the component 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-bordered"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={Object {}}
          >
            <table
              style={
                Object {
                  "tableLayout": "auto",
                }
              }
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": "25%",
                      "width": "25%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "25%",
                      "width": "25%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "25%",
                      "width": "25%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "25%",
                      "width": "25%",
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Amount
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Gateway
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Status
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Created At
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={1}
                  onClick={[Function]}
                  style={Object {}}
                >
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <span>
                      1 INR
                    </span>
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Gateway
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    status
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <span
                      title="2020-12-12"
                    >
                      4 months ago
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        className="ant-pagination ant-table-pagination ant-table-pagination-right"
        style={Object {}}
        unselectable="unselectable"
      >
        <li
          aria-disabled={true}
          className="ant-pagination-prev ant-pagination-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Previous Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="left"
              className="anticon anticon-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </button>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={1}
        >
          <a
            rel="nofollow"
          >
            1
          </a>
        </li>
        <li
          aria-disabled={true}
          className="ant-pagination-next ant-pagination-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Next Page"
        >
          <button
            className="ant-pagination-item-link"
            disabled={true}
            tabIndex={-1}
            type="button"
          >
            <span
              aria-label="right"
              className="anticon anticon-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
